packages/browser/src/browser/index.ts

```diff
@@ -270,6 +271,27 @@ async function registerPlugins(
     )
   }
 
+  // destination plugins
+  await Promise.allSettled(
+    Object.entries(options.destinations ?? {}).map(
+      async ([name, settingsOrDestination]) => {
+        const plugin =
+          settingsOrDestination instanceof Destination
+            ? settingsOrDestination
+            : await createDestination(name, settingsOrDestination)
+
+        if (plugin) {
+          toRegister.push(plugin)
```

Comment:
So it looks like we're still hooking into the vanilla system, by appending the plugin list, e.g. `toRegister.push(<plugin>)` like on line 265. And this works because the new Destination class you created still adheres to the same API as the hightouchio plugin defined here: https://github.com/ht-sdks/events-sdk-js-mono/blob/master/packages/browser/src/plugins/hightouchio/index.ts#L126L138.

My main question then is, why did this require making a new plugin class code as `Destination`? Did the "GTM Destination" plugin-class-meeting-this-API already exist elsewhere, but just in a different repo? How would the code have worked in the before times if someone had used seg GTM as a segment plugin?

Reply:
Correct. 

The GTM plugin for Segment [lives here](https://github.com/segmentio/analytics.js-integrations/blob/master/integrations/google-tag-manager/lib/index.js). When the configuration is fetched on load from the Segment `/settings` API the SDK loads "legacy" destinations like the GTM integration [here](https://github.com/ht-sdks/events-sdk-js-mono/blob/master/packages/browser/src/browser/index.ts#L205) and the newer "actions" destinations [here](https://github.com/ht-sdks/events-sdk-js-mono/blob/master/packages/browser/src/browser/index.ts#L238).

There was a lot of ceremony around the dynamic/remote loading of JS from Segment CDNs so I decided to simplify things by creating the `Destination` class which handles the "plugin" logic and just requires providing the implementations of the tracking methods you want to support.

I probably could've made classes like [`LegacyDestination`](https://github.com/ht-sdks/events-sdk-js-mono/blob/master/packages/browser/src/plugins/ajs-destination/index.ts#L66) work but it seemed like a lot of cruft to bring in for something pretty simple like GTM.

I modeled `Destination` after [`ActionDestination`](https://github.com/ht-sdks/events-sdk-js-mono/blob/master/packages/browser/src/plugins/remote-loader/index.ts#L27) which is Segment's newer way of implementing destinations.